File: src/zeit/ui/tracking_state.py
from typing import Self


class TrackingState:
    icon: str
    status_message: str
    can_toggle: bool

    def __init__(self, icon: str, status_message: str, can_toggle: bool) -> None:
        self.icon = icon
        self.status_message = status_message
        self.can_toggle = can_toggle

    @classmethod
    def not_within_work_hours(cls, status_message: str) -> Self:
        return cls(icon="🌙", status_message=status_message, can_toggle=False)

    @classmethod
    def paused_manual(cls) -> Self:
        return cls(icon="⏸️", status_message="Tracking paused (manual)", can_toggle=True)

    @classmethod
    def active(cls) -> Self:
        return cls(icon="📊", status_message="Tracking active", can_toggle=True)

File: ZeitApp/Sources/ZeitApp/Models/TrackingState.swift
import Foundation

/// Represents the current tracking state of the app
enum TrackingState: Equatable, Sendable {
    case active
    case pausedManual
    case outsideWorkHours(message: String)

    /// SF Symbol name for the menubar icon
    var iconName: String {
        switch self {
        case .active:
            return "chart.bar.fill"
        case .pausedManual:
            return "pause.fill"
        case .outsideWorkHours:
            return "moon.fill"
        }
    }

    /// Status message to display in the menu
    var statusMessage: String {
        switch self {
        case .active:
            return "Tracking active"
        case .pausedManual:
            return "Tracking paused (manual)"
        case .outsideWorkHours(let message):
            return message
        }
    }

    /// Whether the user can toggle tracking on/off
    var canToggle: Bool {
        switch self {
        case .active, .pausedManual:
            return true
        case .outsideWorkHours:
            return false
        }
    }

    /// Whether tracking is currently running
    var isActive: Bool {
        self == .active
    }
}
